src/shared/components/home/admin-settings.tsx

```diff
@@ -60,6 +61,8 @@ export class AdminSettings extends Component<any, AdminSettingsState> {
     this.parseMessage = this.parseMessage.bind(this);
     this.subscription = wsSubscribe(this.parseMessage);
 
+    check_auth();
```

Comment:
Instead of doing this on the client, I think it would  better to handle this on the server and redirect the user before even sending the page that requires auth back.

Reply:
To be honest Im not good at frontend dev. If you want you can take over this PR because it takes me forever just to implement the basics. So Id rather stay in Rust :)